services/efs-guardian/src/simcore_service_efs_guardian/services/notifications/_notifier.py

```diff
@@ -0,0 +1,59 @@
+import contextlib
+from typing import Final
+
+import socketio  # type: ignore[import-untyped]
+from fastapi import FastAPI
+from fastapi.encoders import jsonable_encoder
+from models_library.api_schemas_webserver.socketio import SocketIORoomStr
+from models_library.projects_nodes_io import NodeID
+from models_library.users import UserID
+from pydantic import BaseModel, ByteSize, Field
+from servicelib.fastapi.app_state import SingletonInAppStateMixin
+
+SOCKET_IO_EFS_DISK_USAGE_EVENT: Final[str] = "efsNodeDiskUsage"
+
+
+class EfsNodeDiskUsage(BaseModel):
+    node_id: NodeID
+    used: ByteSize = Field(description="used space")
+    free: ByteSize = Field(description="remaining space")
+    total: ByteSize = Field(description="total space = free + used")
+    used_percent: float = Field(
+        gte=0.00,
+        lte=100.00,
+        description="Percent of used space relative to the total space",
+    )
```

Comment:
Q: how will this data be interpreted by the FE?

Reply:
A: Frontend will reuse the same widget as now, it will just populate it with this data (If EFS enabled)

Reply:
Followup question: this means that you will have metrics on some paths coming from the EFS and for some paths coming from the disk in the sidecar.

One thing to note. This widget is incorrectly displaying only 1 out of 4 paths that it's supposed to provide data from. The fronted can't reuse it as is. and it has to figure out how to display the information properly. At least two different disk quotas need to be displayed here. One for `states` and one for `inputs` + `outputs` + `/tmp`

![Screenshot 2024-10-14 at 11 01 05](https://github.com/user-attachments/assets/8d371f63-2ab2-4929-af71-8253bdf8091a)

Reply:
Note your information will be misleading otherwise

Reply:
fixed